### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by kjerandp**

### CODE REVIEW

File: src/tracks/track.ts [hunk 1]
```diff
@@ -142,16 +145,17 @@ export default class Track {
     }
   }
 
+  onDataLoaded(data?: any) {
+    if (this.isLoading) this.isLoading = false;
+  }
+
```
Comment (Brynjulf) [inline src/tracks/track.ts]: is there supposed to be validation on the presence of data to set loading status?

File: src/ui/track-group-styles.scss [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }
+  .track:last-child {
+    border-right: none;
+  }
+  .overlay {
+    pointer-events: auto;
+    min-height: 0;
+    position: absolute;
+    z-index: 200;
+    // background-color: rgba(127, 255, 127, 0.1);
```
Comment (Brynjulf) [inline src/ui/track-group-styles.scss]: can remove this
```suggestion

```

File: src/tracks/track.ts (hunk 1 above)
Comment (kjerandp) [reply src/tracks/track.ts]: Track is an abstract class. The parameter is included to show its signature

File: src/utils/scale-helper.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        ticks.minor.push(scale.invert(x));
+      }
+    }
+    return ticks;
+  }
+
+  /**
+   * Internal. Creates minor ticks based around a value, number of steps and step size
+   */
+  static createMinorTicks(v: number, steps: number, stepSize: number) : number[] {
```
[the remark this answers is not in the document]
Comment (kjerandp) [reply src/utils/scale-helper.ts]: Made private

File: src/utils/data-helper.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+ */
+  static reduceSegment(segment: PlotData) : Tuplet<number> {
+    const avgV = mean(segment, d => d[1]);
+    const avgD = mean(segment, d => d[0]);
+
+    return [avgD, avgV];
+  }
+
+  /**
+  * Test if the data is withing the scale's domain
```
[the remark this answers is not in the document]
Comment (kjerandp) [reply src/utils/data-helper.ts]: typo fixed

**Pull request merged**